var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -84,4 +112,54 @@ def cmake_args(self):
         else:
             cmake_args.append("-DBUILD_TESTS=OFF")
 
+        cpu_vendor = self.spec.platform
+        cpu_uarch = self.spec.target
+
+        #taken from list of archspec.cpu.TARGETS
+        supported_amd_targets = ["zen2"]
+        supported_arm_targets = ["neoverse_n1"]
+        supported_ibm_targets = ["power9le"]
+        supported_intel_targets = [
+            "sandybridge",
+            "ivybridge",
+            "haswell",
+            "broadwell",
+            "skylake",
+            "cascadelake",
+            "icelake",
+        ]
+
+        if cpu_vendor == "AuthenticAMD" and cpu_uarch in supported_amd_targets:
```

Comment:
These are going to overwrite the values you set above for the gpu architectures. It seems like there are two options based on what I don't know about the package.

Either, you build for both the GPU and the CPU. In that case, change this block to only modify the CPU values, and the block above modifies the GPU values (and needs an `else` case to turn them all off).

If you only build for one, then these should all be `elif` on the same code block as the GPU settings above.